puppet/zulip/templates/nginx/upstreams.conf.template.erb

```diff
@@ -2,10 +2,19 @@ upstream django {
     server unix:/home/zulip/deployments/uwsgi-socket;
 }
 
+<% if @tornado_multiprocess -%>
+<% @tornado_ports.each do |port| -%>
+upstream tornado<%= port %> {
+    server 127.0.0.1:<%= port %>;
+    keepalive 10000;
+}
+<% end -%>
+<% else -%>
 upstream tornado {
     server 127.0.0.1:9993;
     keepalive 10000;
 }
+<% end -%>
```

Comment:
I think we can actually remove the `else` clause here with tiny bit of work to `app_frontend_base.pp` to set `tornado_ports` in the else clause there; it's really just a loop over `tornado_ports`, and that'd be a lot more readable, right?

Reply:
Oh, I guess maybe we can't because the upstream name ends up being e.g. `tornado9803`?

Reply:
Yes exactly, I think we'd need to make no-sharding tornado set up work on port `9800` first. Seems like a worthwhile thing to try to do, for  consistency